tarodb/assets_store.go

```diff
@@ -1226,6 +1254,30 @@ func queryChainAssets(ctx context.Context, q ActiveAssetsStore,
 	return matchingAssets, nil
 }
 
+// FetchCommitment returns a specific commitment identified by the given asset
+// parameters. If no commitment is found, ErrNoCommitment is returned.
+func (a *AssetStore) FetchCommitment(ctx context.Context, id asset.ID,
+	anchorPoint wire.OutPoint, amt uint64,
+	groupKey *asset.GroupKey) (*tarofreighter.AnchoredCommitment, error) {
+
+	filter, err := specificAssetFilter(id, anchorPoint, amt, groupKey)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create filter: %w", err)
+	}
+
+	commitments, err := a.queryCommitments(ctx, filter)
+	if err != nil {
+		return nil, fmt.Errorf("unable to query commitments: %w", err)
+	}
+
+	if len(commitments) != 1 {
```

Comment:
Yeh see other comments re letting more than 1 go thru here, and then instead adding the temporary restriction at a higher level (within the wallet wallet itself).

Reply:
You're right, with the previous query it could've selected two coins of the same asset if they had the same value (or both a value greater than what we need). I replaced the amount with the script key in the query, which should only ever give one result (since the script key needs to be unique per anchor TX, I believe).

@Roasbeef at this level we are looking to get one specific asset input, independent of how many we allow on the virtual TX level.